### PR TITLE
react: frames hooks

### PR DESCRIPTION
**Pull request by krzysu**

### CODE REVIEW

File: examples/web/src/misc/UseSignFrameAction.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    if (!identityToken) {
+      throw new Error('No identity token found');
+    }
+
+    setFrameActionToVerify({
+      signature,
+      signedTypedData: {
+        domain: signedTypedData.domain,
+        types: signedTypedData.types,
+        value: signedTypedData.message, // TODO: this mapping is annoying
```
Comment (krzysu) [inline examples/web/src/misc/UseSignFrameAction.tsx:77]: @cesarenaldi in the react sdk we alias typed data `value` as `message` which is now "disturbing" with frame action verification, is there any meaning behind it? can we remove the alias from all typed data? it has impact on how the signer sees them.

File: examples/web/src/Providers.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ const wagmiConfig = createConfig({
 });
 
 const lensConfig: LensConfig = {
-  environment: development,
+  environment: staging,
```
Comment (krzysu) [inline examples/web/src/Providers.tsx:18]: this is temporary

File: examples/web/src/misc/UseSignFrameAction.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+import { RequireProfileSession } from '../components/auth';
+
+function ServerVerifySignature({ toVerify }: { toVerify: FrameVerifySignature }) {
+  const [verified, setVerified] = useState<FrameVerifySignatureResult>();
+  const storage = useStorage();
+  const client = useMemo(
+    () =>
+      new LensClient({
+        environment: staging,
```
Comment (krzysu) [inline examples/web/src/misc/UseSignFrameAction.tsx:27]: this is temporary only

File: packages/react/src/wallet/adapters/ConcreteWallet.ts [hunk 4]
```diff
@@ -155,6 +158,52 @@ export class ConcreteWallet extends Wallet {
     }
   }
 
+  async signFrameAction<TData extends TypedDataShape>(
```
Comment (krzysu) [inline packages/react/src/wallet/adapters/ConcreteWallet.ts:161]: I can't figure out the type here for `_signTypedData` later to work with `data.domain`, `data.types` etc. below, so I introduced TypedDataShape, let me know what would be a better way, to make TypedData the implementation detail here.

```
signer._signTypedData(
        unsignedAction.data.domain,
        unsignedAction.data.types,
        unsignedAction.data.message,
      );
```

File: .changeset/early-keys-know.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+"@lens-protocol/blockchain-bindings": minor
+"@lens-protocol/api-bindings": minor
+"@lens-protocol/domain": minor
+"@lens-protocol/react": minor
+"@lens-protocol/react-native": minor
+"@lens-protocol/react-web": minor
+---
+
+**feat:** added `useSignFrameAction` hook
```
Comment (cesarenaldi) [inline .changeset/early-keys-know.md:10]: Don't we need a client changeset too?

File: packages/blockchain-bindings/package.json [hunk 6]
```diff
@@ -40,7 +40,6 @@
   "license": "MIT",
   "dependencies": {
     "@ethersproject/abi": "^5.7.0",
-    "@ethersproject/abstract-signer": "^5.7.0",
```
Comment (cesarenaldi) [inline packages/blockchain-bindings/package.json:43]: 👋

**Review by cesarenaldi — APPROVED** [review]
Neat!

File: .changeset/early-keys-know.md (hunk 5 above)
Comment (krzysu) [reply .changeset/early-keys-know.md:10]: I skipped it here as this PR merges into the client SDK one, and there is a changeset for client already there 

https://github.com/lens-protocol/lens-sdk/pull/887/files#diff-270dee66ee89cf46e8101f30e11bc690d809f1bf8943c25c0b4e36e0a7196e5b

**Pull request merged**